aya/src/programs/tc.rs

```diff
@@ -89,6 +97,12 @@ pub enum TcError {
     /// the clsact qdisc is already attached
     #[error("the clsact qdisc is already attached")]
     AlreadyAttached,
+    /// tcx links can only be attached to ingress or egress, custom attachment is not supported
+    #[error("tcx links can only be attached to ingress or egress, custom attachment: {0} is not supported")]
+    InvalidTcxAttach(u32),
+    /// operation not supported for programs loaded via tcx
+    #[error("operation not supported for programs loaded via tcx")]
+    InvalidLinkOperation,
```

Comment:
I would remove obth this and InvalidTcxAttach and return None where necessary
instead. The way things are now, TcAttachType::parent returns "something" but
it's not really a valid value for TCX. Instead of adding another error for that
case, I'd just return None everywhere.

Reply:
As I’ve worked with the code in this pr and in my bpfman implementation I’ve come to the opinion that it would be better to have different attach types for TC and TCX because they are really two different hook points and attach types with different kernel APIs and attributes.  Trying to treat them as the same adds complication as you are pointing out here.  Since I wasn’t involved in the earlier design discussions, what’s implemented works, and it’s kind of late in the game, I was going to wait and bring this up after the pr merges and propose a solution to make it cleaner.

That said, with the current design, I think it’s best to leave this code as it is for now.  Some details follow.

`TcAttachType::tcx_attach_type()` should only be called from a TCX context, and the attach type should never be `Custom`. If the user makes an error, it's better to return an error here rather than requiring the caller to handle a `None` error case later.

`TcAttachType::parent` should only be used in a TC (netlink) context, which it currently is. However, we can't detect if it's mistakenly used in a TCX context so we can’t return something different for TCX. To help clarify, I renamed `parent` to `tc_parent` to indicate that it applies only to TC.